core/che-core-api-core/src/test/java/org/eclipse/che/api/core/util/VersionTest.java

```diff
@@ -0,0 +1,166 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2017 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.core.util;
+
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+
+/**
+ * @author Anatoliy Bazko
+ */
+public class VersionTest {
+    @Test(dataProvider = "testValidVersion")
+    public void testValidVersion(String version) throws Exception {
+        Version.validate(version);
+    }
+
+    @DataProvider(name = "testValidVersion")
+    public static Object[][] testValidVersion() {
+        return new Object[][] {{"0.0.1"},
+                               {"1.0.1"},
+                               {"10.3.0"},
+                               {"10.3.0.0"},
+                               {"10.3.0.1"},
+                               {"10.3.0.1-RC"},
+                               {"0.9.0"},
+                               {"1.0.0"},
+                               {"1.0.10"},
+                               {"1.0.10-RC"},
+                               {"1.0.1-SNAPSHOT"},
+                               {"1.0.1-RC-SNAPSHOT"},
+                               {"1.0.1.0-SNAPSHOT"},
+                               {"1.0.1-M1"},
+                               {"1.0.1.1-M1"},
+                               {"1.0.1-M1-SNAPSHOT"},
+                               {"1.0.1.2-M1-SNAPSHOT"},
+                               {"1.0.1.2-beta-1-SNAPSHOT"},
+                               {"1.0.1.2-M1-beta-1-SNAPSHOT"},
+                               {"1.0.1.2-M1-RC-SNAPSHOT"}};
```

Comment:
Please ensure you haven't missed combinations such as 1.0.1.2-M1-beta-1-RC-SNAPSHOT